### PR TITLE
:seedling: Added PR message verification with kubebuilder-release-tools

### LINKED ISSUE #1548: Enforce some standards for PR titles
/kind feature

**Describe the solution you'd like**
CAPI uses the [kubernetes-sigs/kubebuilder-release-tools](https://github.com/kubernetes-sigs/kubebuilder-release-tools) action to enforce some standards in PR titles. These standards require emoji prefixes to PR titles, which make release note generation and visually scanning through PRs convenient.

**Anything else you would like to add:**
This should be a really straightforward addition (a single workflow), happy to take this up myself or have a new contributor work on it once the issue is validated.

### PR DESCRIPTION
**Pull request by AectannArd**
**What this PR does / why we need it**:
It adds instrumentation for verifying PR with kubebuilder-release-tools

**Which issue(s) this PR fixes**:
Fixes #1548 

**Special notes for your reviewer**:
Brings dependency on kubebuilder-release-tools
No images are affected

**Release note**:

### CODE REVIEW

File: go.mod
Comment (ditsuke) [inline go.mod:136]: These dependencies must go in the `hack/tools` module, then there you can perhaps add a Makefile target to pull down the release note binary and a target in the root makefile to generate notes.

**Review by ditsuke — CHANGES_REQUESTED** [review]
The kubebuilder note generation binary and its dependencies should go to the `hack/tools` module.

**Comment by AectannArd** [comment]
Sure thing. I'll fix both the @ditsuke comment and the markdown issue

File: go.mod
Comment (ditsuke) [reply go.mod:132]: changes to this file need to be reverted

**Comment by AectannArd** [comment]
I've fixed the aforementioned issues.
A small notice should be that if your are planning to use release-note generation then the branching structure of the repo will have to be changed 'cause the note generation tool works only with hard-coded patterns which seem to be not overrideable.

File: go.mod
Comment (AectannArd) [reply go.mod]: done

**Comment by AectannArd** [comment]
@srm09 please take a look

**Review by ditsuke — APPROVED** [review]
LGTM @AectannArd. Apologies for the wait on review

**Comment by AectannArd** [comment]
@ditsuke thanks, let's hope that @srm09 will find a moment to check it in the nearest future

**Comment by srm09** [comment]
@AectannArd Can you suggest what are the changes required for the repo? I'd like to close loop on this one in the next release, if possible and if the changes to the repo structure don't have far reaching effects.

**Comment by AectannArd** [comment]
@srm09 All the file changes for the feature to work are supplied in this PR.
They bring 2 things:
* a github-action for controlling the MR-titles (it will demand that all the following PRs have a classifier-prefix, otherwise the MR will fail to build
* a command `make generate-release-notes` for you to call while preparing for the release which will output the groupped textblocks of release changeset

The worst thing that could happen is that all the PRs open at the moment will have to fix their title.

**Comment by AectannArd** [comment]
@srm09 can we go ahead and merge or should we wait for @vrabbi 's approval?

**Pull request merged**
